airbyte-integrations/connectors/source-snowflake/source_snowflake/source.py

```diff
@@ -49,45 +49,35 @@ def check_connection(self, logger, config) -> Tuple[bool, any]:
 
         except IncorrectHostFormat:
             error_message = f"your host is not ending with {self.SNOWFLAKE_URL_SUFFIX}"
-            emit_airbyte_error_message(error_message=error_message,
-                                       failure_type=FailureType.config_error)
-            raise ValueError(error_message)
 
         except UnknownUpdateMethodError as e:
             error_message = f'Update method {e.unknown_update_method} not recognized'
-            emit_airbyte_error_message(error_message=error_message,
-                                       failure_type=FailureType.config_error)
-            raise ValueError(error_message)
 
         except DuplicatedPushDownFilterStreamNameError:
             error_message = 'There are pushdown filters with same name which is not possible'
-            emit_airbyte_error_message(error_message=error_message,
-                                       failure_type=FailureType.config_error)
-            raise ValueError(error_message)
 
         except InconsistentPushDownFilterParentStreamNameError as e:
             error_message = (f'You have provided inconsistent pushdown filters configuration. Parent stream not found or mistake in  '
                              f'spelling parent stream name. Correct spelling must be your_schema_name.your_table_name and make sure '
                              f'you have the rights to read the table.\n. Here is the '
                              f'list of the streams affected by this error: {e.push_down_filters_without_consistent_parent}')
-            emit_airbyte_error_message(error_message=error_message,
-                                       failure_type=FailureType.config_error)
-            raise ValueError(error_message)
 
         except StopIteration:
             error_message = "There is no stream available for the connection specification provided"
-            emit_airbyte_error_message(error_message=error_message,
-                                       failure_type=FailureType.config_error)
-            raise StopIteration(error_message)
 
         except requests.exceptions.HTTPError as error:
             error_message = error.__str__()
             error_code = error.response.status_code
+
             if int(error_code) == 412:
                 error_message = ("SQL execution error for check.\n"
                                  "The origin of the error is very likely to be:\n"
                                  "- The configuration provided does not have enough permissions to access the requested database/schema.\n"
                                  "- The configuration is not consistent (example: schema not present is database.")
+
+            if int(error_code) == 404:
```

Comment:
It seems very confusing for the person debugging that the HTTP status is `404` but the error message mentions `401`. What is the reason for that?

Reply:
Based on the commit message, it was a typo. How was the test for `secrets/invalid_private_key_config.json` passing then?

Reply:
It was caught by another custom error class (BadPrivateKeyFormatError) I introduced, the error is about deserialization not about having incorrect credential values

Reply:
It is confusing, i was a type from me i corrected it.